unit-tests-ext/src/test/scala/java/nio/file/attribute/FileTimeExtTest.scala

```diff
@@ -0,0 +1,106 @@
+package java.nio.file.attribute
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import org.junit.Test
+import org.junit.Assert._
+
+class FileTimeExtTest {
+  private val timestamp          = 1582230020000L
+  private val timestampAsSeconds = timestamp / 1000L
+  private val timestampAsNanos   = timestamp * 1e6.toLong
+  private val nanosAdjustment    = 1234L
+
+  @Test def fromInstant(): Unit = {
+    def fileTimeFromInstantEquals(ft: FileTime, instant: Instant) = {
+      assertEquals(timestamp, instant.toEpochMilli)
+      assertEquals(timestamp, ft.toMillis)
+
+      val fromInstant = FileTime.from(instant)
+      assertEquals(0, ft.compareTo(fromInstant))
+      assertEquals(timestamp, fromInstant.toMillis)
+      assertEquals(timestampAsSeconds, fromInstant.to(TimeUnit.SECONDS))
+      assertEquals(ft, fromInstant)
+    }
+
+    fileTimeFromInstantEquals(FileTime.fromMillis(timestamp),
+                              Instant.ofEpochMilli(timestamp))
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampAsSeconds, TimeUnit.SECONDS),
+      Instant.ofEpochSecond(timestampAsSeconds))
+
+    fileTimeFromInstantEquals(
+      FileTime.from(timestampAsNanos + nanosAdjustment, TimeUnit.NANOSECONDS),
+      Instant.ofEpochSecond(timestampAsSeconds, nanosAdjustment))
+  }
+
+  @Test def fromInstantMaxPrecision(): Unit = {
```

Comment:
It seems to me that this test is overengineered, to the point where it is difficult to understand what it tests, and how. Consider using "dumb" tests instead, using known input/output pairs and relying on `FileTime.equals`:
```scala
assertEquals(FileTime.from(1L, DAYS), FileTime.from(Instant.ofEpochSecond(86400L, 0)))
```

Reply:
Playing more with that, I think I was wrong before. Storing `value` and `unit` is not enough. Apparently `FileTime` has strictly more precision than `Instant`. I could not find an `Instant` for which `FileTime.from(instant).toInstant() == instant` was false.

So, in fact, it seems the reality is much simpler: a `TimeUnit` stores a pair of `days: Long` and `nanos: Long`. Probably with `nanos` capped at `86400000000000L - 1L` (so working with a `floorMod(_, 86400000000000L)`), the number of nanoseconds in a day, to make things easier.

That would make the implementation of all methods, in particular `equals` and `compareTo`, much easier.

Reply:
Actually `FileTime.from(instant).toInstant() eq instant` also would be true. I belive JDK stores original Instant or created new one based on other parameters.

Reply:
Right, but we can't do that (because a field of type `Instant` would not link when we don't have `Instant` in the codebase), and there's certainly no requirement to offer `eq`. We should be able to guarantee `==`, though.